### ISSUE TITLE
Automatically update template repo ref in renku-ui values

### ISSUE THREAD

**Issue by ableuler** [SwissDataScienceCenter/renku-project-template#89]
Changes to master in this repo could automatically open a PR in the renku-ui repo to change the corresponding line in the values.yaml file of the UI chart. This would ensure that released versions of the ui chart use the up-to-date version of that repo. What do you think @SwissDataScienceCenter/renku-ui-maintainers?

**Comment by rokroskar** [comment]
I think that value should be set in the main values file, since it's a deployment setting.

**Comment by rokroskar** [comment]
The UI could optionally provide a maximum (minimum) template version that it supports/requires.

**Comment by ableuler** [comment]
> I think that value should be set in the main values file, since it's a deployment setting.

Makes sense. My point was mostly about about automatically using the head of this repo in our dev deploymentl. I guess we could automatically update the renku values if we want this, but maybe this repo is not changing frequently enough to justify the effort.

**Comment by rokroskar** [comment]
I _think_ our default should be `master` of the template repo. Then we don't have to update anything very frequently and we can just set the specific value when we do a release so it's stable when people deploy it.

**Comment by lorenzo-cavazzi** [comment]
I believe the UI is now flexible enough to support a broad range of changes in the templates without requiring an update, so it makes sense to move this value setting somewhere else. The main values file seems a reasonable choice.

> I think our default should be master of the template repo.

If you mean that `master` should be the default in renku-ui but _always_ overwritten in 3rd pary RenkuLab deployment, I agree. (but this is not different now, we overwrite that value in all our deployments).
Otherwise, we would need to pay extreme care at possible breaking changes (even though they should be rare). Imagine a breaking change going in that would suddenly break all the RenkuLab deployments that are not up to date.
Another minor detail is that it's very easy to create identical projects, you just need to provide the same tag to the renku init command. If the target tag becomes `master`, that won't be trivial anymore.

**Comment by rokroskar** [comment]
Sure, what I'm proposing is not any different than the default value in a development chart pointing to a "latest" image and then released charts pointing to a specific, pinned version. It's just that atm the value of the templates tag gets updated sporadically in the UI chart which makes things largely inconsistent. If there is, at some point, a reason to enforce a specific tag within a UI stream of development, we can of course pin it in the UI chart itself to be on the extra safe side.
